compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -416,7 +418,8 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
         val flagName = LazyBitMapName.fresh("0".toTermName)
         val flagSymbol = newSymbol(claz, flagName, containerFlags, defn.LongType).enteredAfter(this)
         flag = ValDef(flagSymbol, Literal(Constant(0L)))
-        val offsetTree = ValDef(offsetSymbol.nn, getOffset.appliedTo(thizClass, Literal(Constant(flagName.toString))))
+        val fieldTree = thizClass.select(lazyNme.RLazyVals.getDeclaredField).appliedTo(Literal(Constant(flagName.toString)))
```

Comment:
We would have to fallback to the old scheme when compiling with `scalaOutputVersion` set to `3.0` or `3.1`